salt/metalk8s/defaults.yaml

```diff
@@ -4,6 +4,8 @@ kubeadm_preflight:
       - util-linux            # provides nsenter, mount
       - iproute               # provides ip
       - iptables              # provides iptables
+      - m2crypto              # provides x509 salt support
```

Comment:
Not really a dep for `preflight`, which is what this pillar value refers to. Could we keep dependencies closer to the code/state/... that actually needs them and install them there?

Reply:
For me `preflight` is what we need to have to run kubeadm and for `python-netaddr` currently it's used when render the state so we need it available before rendering the state.

Reply:
Not sure... There'll be other places where we need this library to be installed as well.

I wouldn't stick to the kubeadm layout too much. Honestly, not even sure why this list is in some Yaml file somewhere and not in the `preflight` state SLS directly.

In general, I'd install dependencies in the SLS that requires them, if possible, or in whatever state deploys something that needs it (e.g. if we deploy a cert validity beacon at some point).

Reply:
I'd say in preflight we may want to check if dependencies are available, but without installing them.
The only packages we should install in preflight are the ones needed to perform some checks.

So in this case, we may not want to install `m2crypto` but we may want to check if it's available for install.